packages/framework/agent-scheduler/src/scheduler.ts

```diff
@@ -387,20 +387,22 @@ export class AgentScheduler extends TypedEventEmitter<IAgentSchedulerEvents> imp
 }
 
 class AgentSchedulerRuntime extends FluidDataStoreRuntime {
-    private readonly agentSchedulerP: Promise<AgentScheduler>;
     constructor(
         dataStoreContext: IFluidDataStoreContext,
         sharedObjectRegistry: ISharedObjectRegistry,
         existing: boolean,
     ) {
-        super(dataStoreContext, sharedObjectRegistry, existing);
-        this.agentSchedulerP = AgentScheduler.load(this, dataStoreContext, existing);
+        super(
+            dataStoreContext,
+            sharedObjectRegistry,
+            existing,
+            async () => AgentScheduler.load(this, dataStoreContext, existing));
     }
     public async request(request: IRequest) {
         const response = await super.request(request);
         if (response.status === 404) {
             if (request.url === "" || request.url === "/") {
-                const agentScheduler = await this.agentSchedulerP;
+                const agentScheduler = await this.IFluidHandle?.get();
```

Comment:
this could technically be undefined, which would break the request contract. my suggestion around defaulting to the request in the datastore runtime is interesting in this case. it would allow handle to always be defined, but it could also result in reentrancy and infinite recursion if we did fallback, and then the handle was retrieved in request.  given this, i think the default is probably a bad idea, and dealing with undefined is fine.

Reply:
I did run into the reentrancy issue with DS factories (lazy loaded, test fluid object) when forwarding the request to the runtime if the entry point object couldn't handle it or was undefined. Since the current version of the code can synchronously guarantee that the objects that handle the request are defined, and we'd expect that to still be the case with the new code, I just went with asserts. If we missed some case/flow this might introduce a behavior change if the asserts "trigger", but if so I think we treat that as a bug and fix it.